### PR TITLE
[Poloniex] Added order status to Poloniex

### PR DESCRIPTION
**Pull request by traviscollins**
I've implemented an imperfect, but best possible order status for Poloniex. Poloniex API frustratingly does not return the order type, or the originalAmount value for orders. But the rest of it is there.

### CODE REVIEW

File: xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java [hunk 1]
```diff
@@ -313,4 +316,59 @@ public static ExchangeMetaData adaptToExchangeMetaData(
...
+    OrderType orderType = null;
+    CurrencyPair currencyPair = null;
+    BigDecimal amount = new BigDecimal(0);
+    BigDecimal fee = new BigDecimal(0);
+
+    List<BigDecimal> prices = new ArrayList<>();
+
+
+    //"2018-06-12 19:07:01"
+    SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");
```
Comment (walec51) [inline xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java]: make this a static

File: xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java [hunk 2]
```diff
@@ -313,4 +316,59 @@ public static ExchangeMetaData adaptToExchangeMetaData(
...
+
+    for (PoloniexUserTrade poloniexUserTrade : poloniexUserTrades) {
+      id = poloniexUserTrade.getTradeID();
```
Comment (walec51) [inline xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java]: orderId == last trade id
doesn't seam true

File: xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java [hunk 3]
```diff
@@ -313,4 +316,59 @@ public static ExchangeMetaData adaptToExchangeMetaData(
...
+      orderType = poloniexUserTrade.getType().equals("buy") ? OrderType.BID : OrderType.ASK; // what about others?
+
+      try {
+        date = format.parse(poloniexUserTrade.getDate());
```
Comment (walec51) [inline xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java]: order date = last trade date
doesn't seam true

File: xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java [hunk 4]
```diff
@@ -313,4 +316,59 @@ public static ExchangeMetaData adaptToExchangeMetaData(
...
+      amount = amount.add(poloniexUserTrade.getAmount());
+      fee = fee.add(poloniexUserTrade.getFee());
+
+      prices.add(poloniexUserTrade.getRate());
+
+    }
+
+
+
+    BigDecimal averagePrice = prices.stream().reduce(new BigDecimal(0), (a,b) -> a.add(b)).divide(new BigDecimal(poloniexUserTrades.length));
```
Comment (walec51) [inline xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/PoloniexAdapters.java]: this is wrong, you need to take each trades amount into consideration to calculate the proper average

File: xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/service/PoloniexTradeService.java [hunk 5]
```diff
@@ -265,11 +262,35 @@ public OpenOrdersParams createOpenOrdersParams() {
...
+
+    returnValue.addAll(
+            orderIdList.stream()
+                    .filter( f-> !returnValue.stream().filter( a-> a.getId().equals(f)).findFirst().isPresent())
+                    .map(
+                            f-> {
+                              try {
+                                return PoloniexAdapters.adaptUserTradesToOrderStatus(returnOrderTrades(f));
+                              } catch (IOException e) {
+                                e.printStackTrace();
```
Comment (walec51) [inline xchange-poloniex/src/main/java/org/knowm/xchange/poloniex/service/PoloniexTradeService.java]: logging

**Comment by traviscollins** [comment]
@walec51 Fair criticism. See above changes for fixes. I've committed a few times to this project, but havn't seen a style guide.

**Comment by timmolter** [comment]
@traviscollins The style guide is here: https://github.com/knowm/XChange/wiki/Code-Style See also: https://github.com/knowm/XChange/wiki/New-Implementation-Best-Practices

**Pull request merged**